tests/Directory.Build.targets

```diff
@@ -25,11 +25,15 @@
     <Error Condition="'$(TestArchiveTestsDir)' == ''" Text="TestArchiveTestsDir property to archive the test folder must be set." />
     <PropertyGroup>
       <TestsArchiveSourceDir Condition="'$(TestsArchiveSourceDir)' == ''">$(OutDir)</TestsArchiveSourceDir>
+      <ZipTestArchiveTfm></ZipTestArchiveTfm>
+      <ZipTestArchiveTfm Condition="'$(TargetFramework)' != '$(DefaultTargetFramework)'">-$(TargetFramework)</ZipTestArchiveTfm>
     </PropertyGroup>
 
-    <MakeDir Directories="$(TestArchiveTestsDir)" />
-    <ZipDirectory SourceDirectory="$(TestsArchiveSourceDir)"
-                  DestinationFile="$([MSBuild]::NormalizePath('$(TestArchiveTestsDir)', '$(MSBuildProjectName).zip'))"
+    <!-- TestsArchiveSourceDir is empty (OutDir) for multi-target build step -->
+    <MakeDir Condition="$(TestsArchiveSourceDir) != ''" Directories="$(TestArchiveTestsDir)" />
```

Comment:
If it is empty then how why doesn't `ZipDirectory` fail with empty `SourceDirectory`?

Reply:
oh, you are skipping it in that case. I'll do a local build, and look at the binlog.

Reply:
It failed before these changes. Now this is skipped when it is the case, when it's invoking the cross-target build.

Reply:
You can use `and '$(IsInnerBuild)' == 'true'` in the condition for the target. This will be true when it is doing the inner build - which is one per target framework. And the target will get skipped for the outer build.

Reply:
Thanks, will try that instead